### PR TITLE
Command Autocompletion

### PR DESCRIPTION
**Pull request by WilliamMcCumstie**
I have added auto-completion back into metalware. This is done by reading the commands directory structure and generating the auto-complete from that. It does not current support auto completion of options.

In order to do this, I needed to move the `base_command` and `bash` out of the commands directory as they are not commands themselves but used by other commands.

### CODE REVIEW

File: etc/profile.d/base.sh [hunk 1]
```diff
@@ -16,23 +16,29 @@ metal() {
...
-#         cmds=$(ls /opt/metalware/lib/actions)
+        if [ -d "$cur_dir" ]; then
+            cmds=$(ls $cur_dir | sed s/\.rb//g)
+        fi
 
-#         COMPREPLY=( $(compgen -W "$cmds" -- "$cur") )
-#     }
+        # Additional bash commands, ideally these will be migrated to ruby
+        if [ "$path" == "/" ]; then
+            cmds="$cmds power console"
```
Comment (BobWhitelock) [inline etc/profile.d/base.sh]: These commands aren't being autocompleted for me after a fresh install - do they complete for you?

**Comment by BobWhitelock** [comment]
Another thing I've just noticed: if you now enter a command like `metal render /va<tab>` it gives the following:
```
[root@controller(dev) metalware]# metal render /vased: -e expression #1, char 18: unknown option to `s'
r/
```

**Comment by WilliamMcCumstie** [comment]
Rebased this branch on the fixes to the install process. Merge `fix/install` first.

https://github.com/alces-software/metalware/pull/66

**Comment by WilliamMcCumstie** [comment]
Rebased on develop

**Comment by BobWhitelock** [comment]
LGTM

**Pull request merged**